Mage.Sets/src/mage/cards/g/GerrardsHourglassPendant.java

```diff
@@ -0,0 +1,129 @@
+
+    public GerrardsHourglassPendant(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{1}");
+        addSuperType(SuperType.LEGENDARY);
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        // If a player would begin an extra turn, that player skips that turn instead.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GerrardsHourglassPendantSkipExtraTurnsEffect()));
+
+        // {4}, {T}, Exile Gerrard's Hourglass Pendant: Return to the battlefield tapped all artifact, creature,
+        // enchantment, and land cards in your graveyard that were put there from the battlefield this turn.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new GerrardsHourglassPendantReanimateEffect(), new GenericManaCost(4));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new ExileSourceCost());
+        ability.addWatcher(new CardsPutIntoGraveyardWatcher());
+        this.addAbility(ability);
+    }
+
+    private GerrardsHourglassPendant(final GerrardsHourglassPendant card) {
+        super(card);
+    }
+
+    @Override
+    public GerrardsHourglassPendant copy() {
+        return new GerrardsHourglassPendant(this);
+    }
+}
+
+class GerrardsHourglassPendantSkipExtraTurnsEffect extends ReplacementEffectImpl {
+
+    public GerrardsHourglassPendantSkipExtraTurnsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Detriment);
+        staticText = "If a player would begin an extra turn, that player skips that turn instead";
+    }
+
+    public GerrardsHourglassPendantSkipExtraTurnsEffect(final GerrardsHourglassPendantSkipExtraTurnsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GerrardsHourglassPendantSkipExtraTurnsEffect copy() {
+        return new GerrardsHourglassPendantSkipExtraTurnsEffect(this);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Player player = game.getPlayer(event.getPlayerId());
+        MageObject sourceObject = game.getObject(source);
+        if (player != null && sourceObject != null) {
+            game.informPlayers(sourceObject.getLogName() + ": Extra turn of " + player.getLogName() + " skipped");
+        }
+        return true;
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.EXTRA_TURN;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        return true;
+    }
+
+}
+
+class GerrardsHourglassPendantReanimateEffect extends OneShotEffect {
+
+    private static final FilterCard filter = new FilterPermanentCard();
+
+    static {
+        filter.add(PutIntoGraveFromBattlefieldThisTurnPredicate.instance);
+        filter.add(Predicates.not(CardType.PLANESWALKER.getPredicate()));
```

Comment:
this should just have the stated card types inclusively rather than excluding planeswalker, otherwise it can't bring back [Grist, the Hunger Tide](https://scryfall.com/card/mh2/202/grist-the-hunger-tide) and if a hypothetical new permanent type were added it in the future it would bring that back when it shouldn't

Reply:
Done